test/parallel/test-cluster-prof-and-logfile.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+const common = require('../common');
+const tmpdir = require('../common/tmpdir');
+
+const fs = require('fs');
+const assert = require('assert');
+const cluster = require('cluster');
+
+const TEST_TIMEOUT = 1000;
+
+const timeout = ms => new Promise(resolve => setTimeout(resolve, ms));
+const waitForEvent = (eventEmiter, event) => new Promise(resolve => eventEmiter.on(event, resolve));
+
+function findLogFileWithName(name, dir) {
+  const data = fs.readdirSync(dir, 'utf8');
+  return data.filter(file => file.includes(name));
+}
+
+async function shouldFillLogfileArg() {
+  tmpdir.refresh();
+
+  cluster.setupMaster({
+    exec: __dirname,
+    execArgv: ['--prof', '--logfile'],
+    cwd: tmpdir.path
+  });
+
+  cluster.fork();
+  
+  const handler = common.mustCall(worker => {
+    const EXPECTED_FILENAME = 'v8-' + worker.process.pid;
+    const reports = findLogFileWithName(EXPECTED_FILENAME, tmpdir.path);
+    assert.strictEqual(reports.length, 1);
+  });
+
+  cluster.on('exit', handler);
+
+  try {
+    await Promise.race([
+      timeout(TEST_TIMEOUT),
+      waitForEvent(cluster, 'exit')
```

Comment:
This test would be easier to understand without your ad-hoc promisification code,  and be more concise. Lines 38-47 can be replaced with a single line:
```
cluster.once('exit', common.mustCall(() => tmpdir.refresh()));
```
- race - unnecessary, let the test handle timeouts and use common.mustCall
- removeListener - unnecessare, use EventEmitter.once
- try, await, Promise.race, finally - all unnecessary

Reply:
await is unnecessary, because we are working with the same tmp dir